### ISSUE TITLE
Configure token service endpoint

### ISSUE THREAD

**Issue by ThilinaManamgoda** [goharbor/harbor#12815]
As per the [Customize the Harbor Token Service](https://goharbor.io/docs/1.10/install-config/customize-token-service/), it is possible to generate a token according to the spec [Docker Registry v2 authentication](https://github.com/docker/distribution/blob/master/docs/spec/auth/token.md). As per the spec [Docker Registry v2 authentication](https://github.com/docker/distribution/blob/master/docs/spec/auth/token.md)  the registry should return a 401 Unauthorized HTTP response with information on how to authenticate when the registry API is invoked. To enable this process, it is required to configure a token service endpoint configuration on the Harbor registry. But, I couldn't find any documentation to configure a token service endpoint configuration on the Harbor registry.

**Comment by ywk253100** [comment]
I think what the doc `Customize the Harbor Token Service` says is how to customized the certificate and private key used by the token service, not to customized the token service URL

**Comment by ThilinaManamgoda** [comment]
@ywk253100 Yes, this configuration is valid as long as the client knows token service endpoint and scope. But the Docker CLI should be redirected to token service.

**Comment by ywk253100** [comment]
Harbor doesn't support to configure the external token service currently

**Comment by ThilinaManamgoda** [comment]
@ywk253100 Thanks. Is adding the option to configure external token service, part of the Harbor road map?

**Comment by kautkata** [comment]
I'm looking at this issue as it's closed what I can find related to token service internals. What I can't figure out from documentations is how do you actually customize private key used by token service. Documentation is a bit cryptic 

```
cd config/ui
cp /root/cert/private_key.pem private_key.pem
cp /root/cert/root.crt ../registry/root.crt
```
where do you place this stuff ?  in `common/config/registry` directory right beside `config.yml`, `passwd` and empty `root.crt` ?

**Comment by mol-george** [comment]
This is a very useful feature. Is it on the roadmap?
